LumpNav/LumpNav.py

```diff
@@ -843,101 +856,173 @@ def getViewNode(self, viewName):
-      self.cameraYPosSlider.connect('valueChanged(double)', self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewSetCameraYPosMm)
-      self.cameraZPosSlider.connect('valueChanged(double)', self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewSetCameraZPosMm)
-    self.cameraViewAngleSlider.setDisabled(disable)
-    self.cameraXPosSlider.setDisabled(disable)
-    self.cameraZPosSlider.setDisabled(disable)
-    self.cameraYPosSlider.setDisabled(disable)
-    if disable:
-      self.cameraViewAngleSlider.disconnect('valueChanged(double)', self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewSetCameraViewAngleDeg)
-      self.cameraXPosSlider.disconnect('valueChanged(double)', self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewSetCameraXPosMm)
-      self.cameraYPosSlider.disconnect('valueChanged(double)', self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewSetCameraYPosMm)
-      self.cameraZPosSlider.disconnect('valueChanged(double)', self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewSetCameraZPosMm)
     
   def onRightCameraButtonClicked(self):
     logging.debug("onRightCameraButtonClicked {0}".format(self.rightCameraButton.isChecked()))
     viewNode = self.getViewNode('View2')
-    self.viewpointLogic.changeCurrentViewNode(viewNode)
-    if (self.rightCameraButton.isChecked()== True):
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].setViewNode(viewNode)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewSetTransformNode(self.cauteryCameraToCautery)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewStart()
-      self.setDisableSliders(False, viewNode)
-    else:
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewStop()
-      self.setDisableSliders(True, viewNode)
-    self.updateDisableForButtons()
+    self.onCameraButtonClicked(viewNode)
+
+  def onCenterCameraButtonClicked(self):
+    logging.debug("onCenterCameraButtonClicked {0}".format(self.centerCameraButton.isChecked()))
+    viewNode = self.getViewNode('View3')
+    self.onCameraButtonClicked(viewNode)
 
   def onLeftCameraButtonClicked(self):
     logging.debug("onLeftCameraButtonClicked {0}".format(self.leftCameraButton.isChecked()))
     viewNode = self.getViewNode('View1')
+    self.onCameraButtonClicked(viewNode)
+    
+  def onCameraButtonClicked(self, viewNode):
+    logging.debug("onCameraButtonClicked")
     self.viewpointLogic.changeCurrentViewNode(viewNode)
-    if (self.leftCameraButton.isChecked() == True):
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].setViewNode(viewNode)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewSetTransformNode(self.cauteryCameraToCautery)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].trackViewStart()
-      self.setDisableSliders(False, viewNode)
+    if (self.viewpointLogic.nodeInstanceDictionary[viewNode].isCurrentModeTRACKVIEW()):

```

Comment:
What keeps this 'nodeInstanceDictionary' up-to-date? How do we know that there is a member for 'viewNode'?
Probably you want to use a method instead (that can internally handle the case when nodeInstanceDictionary doesn't contain viewNode).